### PR TITLE
Fix lost ReturnUrl param after changing localization

### LINKED ISSUE #16267: Notification editor doesn't close when saving a draft
### Describe the bug
There is an issue with updating content item: when there is a change content item with current culture changing the notification editor doesn't close.

### Orchard Core version

1.8.2.

### To Reproduce
1. Go to Manage Content page
2. Click Add New Global Notification
3. Set some Html text and Content localization to English
4. Еhere is a parameter "returnUrl" in the address bar
5. Press Publish button
6. On Manage Content page select published notification for editing
7. Change Current culture to other language (in this step we are losing a parameter "returnUrl") and change Html text
8. Press Publish button again

### Expected behavior
The editor window should close and return to Manage Content page.

### Logs and screenshots
Issue_close_window_1
Issue_close_window_2
Issue_close_window_3
Issue_close_window_4
Issue_close_window_5

### PR DESCRIPTION
**Pull request by maksim-markin**
Fixes #16267

### CODE REVIEW

**Comment by maksim-markin** [comment]
Related Issue: https://github.com/OrchardCMS/OrchardCore/issues/16267

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/AdminController.cs [hunk 1]
```diff
@@ -82,6 +82,12 @@ public async Task<IActionResult> Localize(string contentItemId, string targetCul
             {
                 var newContent = await _contentLocalizationManager.LocalizeAsync(contentItem, targetCulture);
                 await _notifier.InformationAsync(H["Localized version of the content created successfully."]);
+
+                if(!string.IsNullOrEmpty(returnUrl))
+                {
+                    TempData["returnUrl"] = returnUrl;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/AdminController.cs:88]: Shouldn't it be part of the redirect url instead? added as an argument in the route.

File: src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs [hunk 2]
```diff
@@ -425,6 +425,11 @@ public async Task<IActionResult> Edit(string contentItemId)
 
             var model = await _contentItemDisplayManager.BuildEditorAsync(contentItem, this, false);
 
+            if(!string.IsNullOrEmpty(returnUrl))
+            {
+                TempData["returnUrl"] = returnUrl;
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs:430]: If it's for the View only then ViewData is better. I am worried that you are using TempData but we never had to use it before to support return urls.

**Comment by hishamco** [comment]
@maksim-markin please accept CLA

**Comment by maksim-markin** [comment]
@dotnet-policy-service agree

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/AdminController.cs (hunk 1 above)
Comment (maksim-markin) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/AdminController.cs]: I I simplified returnUrl implementation
commit: 771d500f16db2ab3fad239ef79a311f1125416d9

File: src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs (hunk 2 above)
Comment (maksim-markin) [reply src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs]: I I simplified returnUrl implementation
commit: 771d500f16db2ab3fad239ef79a311f1125416d9

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.SummaryAdminLinks.cshtml [hunk 3]
```diff
@@ -7,16 +7,19 @@
             <span>@T["Localizations"]</span>
         </button>
         <ul class="dropdown-menu dropdown-menu-end">
+            @{
+                var returnUrl = $"{Context.Request.Path}{Context.Request.QueryString}";
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.SummaryAdminLinks.cshtml:11]: Please move it to the top of the file, also no need to string interpolation

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/Edit.cshtml [hunk 4]
```diff
@@ -8,11 +8,13 @@
     ContentItem contentItem = Model.ContentItem;
     var contentTypeDefinition = await ContentDefinitionManager.GetTypeDefinitionAsync(contentItem.ContentType);
     var typeDisplayName = contentTypeDefinition?.DisplayName ?? contentItem.ContentType.CamelFriendly();
+
+    var returnUrl = Context.Request.Query["returnUrl"];
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/Edit.cshtml:12]: Please revert this change coz there's one occurrence in this file

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.SummaryAdminLinks.cshtml (hunk 3 above)
Comment (maksim-markin) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Views/LocalizationPart.SummaryAdminLinks.cshtml]: done

File: src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/Edit.cshtml (hunk 4 above)
Comment (maksim-markin) [reply src/OrchardCore.Modules/OrchardCore.Contents/Views/Admin/Edit.cshtml]: done

**Review by hishamco — APPROVED** [review]

**Comment by hishamco** [comment]
@sebastienros do you have any feedback on this or shall I merge?

**Comment by maksim-markin** [comment]
@sebastienros, @hishamco what about merge?

**Review by sebastienros — APPROVED** [review]
